lunabot_embedded/include/lunabot_embedded/sensor_proc.h

```diff
@@ -0,0 +1,68 @@
+#include <algorithm>
+#include <math.h>
+#include <stdarg.h>
+#include <stdlib.h>
+#include <sys/ioctl.h>
+#include <termios.h>
+#include <array>
+
+using namespace std;
+
+#define DEG2RAD(x) ((x)*M_PI / 180.0)
+
+template <size_t N>
+struct RingBuffer {
+
+  void push(float value) {
+    sum -= data[head];
+    sum += value;
+    data[head] = value;
+    head = (head + 1) % data.size();
+  }
+
+  float mean() { return sum / data.size(); }
+
+  std::array<float,N> data;
+  int head;
+  float sum;
+};
+
+float u_mod(float n, float base) {
+  float m = fmod(n, base);
+  return (m >= 0) ? m : m + base;
+}
+
+// TODO: Fix cyclic spikes in delta
+float deg_angle_delta(float deg, float prev_deg) {
+  float raw = deg - prev_deg;
```

Comment:
Im not fully sure if the logic here is correct. It seems ok but also more complicated than needed. Use this instead
```
diff = deg - prev_deg
diff = (diff + 180) % 360 - 180
```
(angle between -180 and 180)

Reply:
to clarify, your mod is a fmod right? and it assumes that the mod can be negative?

Reply:
nvm. this assumes that the mod is unsigned

Reply:
tested this and it works! Thanks @ashviniyer21!

Reply:
also fixes the spike issue. might be some floating point issue, kinda weird